compiler/data/class-members.h

```diff
@@ -88,6 +89,7 @@ struct ClassMemberInstanceField {
   const TypeHint *type_hint{nullptr};  // from @var / php 7.4 type hint / default value
   int8_t serialization_tag = -1;
   bool serialize_as_float32{false};
+  kphp_json::FieldTags *const kphp_json_tags{nullptr};
```

Comment:
Making this field not const, from the one hand, allows you to call `parse_tag` and others while parsing phpdoc.
From the other hand, parsing phpdoc could be rewritten like this in `ParseAndApplyPhpDocForClass`:
```
if (klass->kphp_json_tags || (f.phpdoc && f.phpdoc.has_tag(PhpDocType::kphp_json)) {
   kphp_json::FieldTags *field_tags = new kphp_json::FieldTags;
   field_tags->parse_from_phpdoc(f.phpdoc);
   field_tags->merge_with(klass->kphp_json_tags, f.local_name());
   f.kphp_json_tags = field_tags;
}
```
This seems simplier at usage, allows making `const` in `ClassMemberInstanceField` (as it definitely should be) and allocates memory only when necessary, leaving `nullptr` for non-jsoned fields.

Same for classes.

Reply:
Hm, nice trade off, I like it. Fixed